kernelci/api/helper.py

```diff
@@ -45,7 +45,9 @@ def receive_event_data(self, sub_id):
 
     def get_node_from_event(self, event_data):
         """Listen for an event and get the matching node object from it"""
-        return self.api.get_node(event_data['id'])
+        if 'id' in event_data:
```

Comment:
IMO the more subtle way of fixing these kind of issues is to verify the event payload before publishing it from the API rather than fixing the parsing.